hash/hash_comparator.go

```diff
@@ -271,12 +278,59 @@ func prepareHashToCompare(splitHash [][]byte, password []byte) (part, salted []b
 		return splitHash[1], md5HashPassword(password), nil
 	}
 
-	return nil, nil, ErrLegacyFandomUnsupportedHash
-	// this one we cannot easily support since we don't know Fandom User ID
-	//
-	// salt := strconv.FormatInt(userID, 10)
-	// return splitHash[0], md5HashPasswordWithSalt(password, salt)
+	userId, err := getCommunityPlatformUserId(identityId)
+	if err != nil {
+		return nil, nil, err
+	}
+	log.Default().Printf("Generating legacy hash for identity: %s", identityId.String())
+	return splitHash[0], md5HashPasswordWithSalt(password, []byte(userId)), nil
+}
+
+type SingleMapping struct {
+	IdentityId *string `json:"identityId,omitempty"`
+	UserId *string `json:"userId,omitempty"`
+}
+
+func getCommunityPlatformUserId(identityId uuid.UUID) (userId string, err error) {
+	// Allow to override service path for local setup
+	serviceUrl, isConfigured := os.LookupEnv("IDENTITY_MAPPER_URL")
+	if !isConfigured {
+		serviceUrl = "http://identity-mapper"
+	}
+	req, err := retryablehttp.NewRequest("GET", serviceUrl + "/appId/community_platform/" + identityId.String(), []byte(""))
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Wikia-Internal-Request", "1")
 
+	client := retryablehttp.NewClient()
```

Comment:
usually it would be best to keep client in memory and do not generate it every time hash check is performed but since those will not be here for long I think we can keep it for sake of simplicity. If any issues with memory arise we can revisit this.